packages/flutter/lib/src/services/deferred_component.dart

```diff
@@ -0,0 +1,157 @@
+  /// The deferred component has previously been installed and all files/assets
+  /// are available, but has not been loaded into the VM in the current
+  /// app's session. Either `loadLibrary()` or [DeferredComponent.installDeferredComponent]
+  /// should be called to trigger the loading process to ensure the deferred component's
+  /// components can be safely used in the app.
+  installedPendingLoad,
+  /// The deferred component has been succesfully downloaded and installed. Assets and
+  /// code from the deferred component are ready to use.
+  installed,
+  /// An installation request has been asked to be canceled.
+  cancelling,
+  /// An installation request has been canceled.
+  canceled,
+  /// An installation request has failed.
+  failed,
+  /// The deferred component has not been requested nor installed or is unavailable.
+  unknown,
+}
+
+/// Manages the installation and loading of deferred component modules.
+///
+/// Deferred components allow Flutter applications to download precompiled AOT
+/// dart code and assets at runtime, reducing the install size of apps and
+/// avoiding installing unnessecary code/assets on end user devices. Common
+/// use cases include deferring installation of advanced or infrequently
+/// used features and limiting locale specific features to users of matching
+/// locales.
+class DeferredComponent {
+  // This class is not meant to be instantiated or extended; this constructor
+  // prevents instantiation and extension.
+  // ignore: unused_element
+  DeferredComponent._();
+
+  // TODO(garyq): We should eventually expand this to install modules by loadingUnitId
+  // as well as moduleName, but currently, loadingUnitId is opaque to the dart code
+  // so this is not possible. The API has been left flexible to allow adding
+  // loadingUnitId as a parameter.
+
+  /// Requests that an assets-only deferred component identified by the [moduleName]
+  /// be downloaded and installed.
+  ///
+  /// This method returns a Future<void> that will complete when the feature is
+  /// installed and any assets are ready to used. When an error occurs, the
+  /// future will complete an error.
+  ///
+  /// This method should be used for asset-only deferred components. Deferred components
+  /// containing dart code should call `loadLibrary()` on a deferred imported
+  /// library's prefix to ensure that the dart code is properly loaded as
+  /// `loadLibrary()` will provide the loading unit id needed for the dart
+  /// library loading process. For example:
+  ///
+  /// ```dart
+  /// import 'split_module.dart' deferred as SplitModule;
+  /// ...
+  /// SplitModule.loadLibrary();
+  /// ```
+  ///
+  /// This method will not load associated dart libraries contained in the dynamic
+  /// feature module, though it will download the files necessary and subsequent
+  /// calls to `loadLibrary()` to load will complete faster.
+  static Future<void> installDeferredComponent({@required String? moduleName}) async {
```

Comment:
moduleName is nullable? The docs above should mention what a null value here means.

Reply:
I'll make this non-nullable for now, The nullability was to accommodate future extensions of this API that may also take a loading unit id. Since this can be added in later in a non-breaking manner, I'll remove the nullability here.